File: OJ practices/LeetCode题目集/70. Climbing Stairs(easy)-动态规划法2.java
/**
 * @Date    : 2019-02-17 22:05:57
 * @Author  : 酸饺子 (changzheng300@foxmail.com)
 * @Link    : https://github.com/SourDumplings
 * @Version : $Id$
 *
 * https://leetcode.com/problems/climbing-stairs/
*/

class Solution
{
    public int climbStairs(int n)
    {
        int res = 0;
        if (n < 3)
        {
            res = n;
        }
        else
        {
            int first = 1, second = 2;
            for (int i = 3; i <= n; ++i)
            {
                res = first + second;
                first = second;
                second = res;
            }
        }
        return res;
    }
}

File: OJ practices/LeetCode题目集/70. Climbing Stairs(easy)-动态规划法2.py
'''
@Author: SourDumplings
@Date: 2020-02-15 23:06:48
@Link: https://github.com/SourDumplings/
@Email: changzheng300@foxmail.com
@Description: https://leetcode-cn.com/problems/climbing-stairs/
'''


class Solution:
    def climbStairs(self, n: int) -> int:
        first = 1
        res = second = 2
        if n < 3:
            return n
        count = 2
        while True:
            res = first + second
            count += 1
            if count == n:
                break
            first = second
            second = res
        return res
